CRM/Activity/BAO/Activity.php

```diff
@@ -933,7 +933,7 @@ public static function deprecatedGetActivities($input) {
     // fixed_sort_order field
     $sql .= "
           UNIQUE KEY ( activity_id )
-        ) ENGINE=HEAP DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci
+        ) ENGINE=HEAP DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci
```

Comment:
So I'm wanting to completely remove this fn for 5.12 or 5.13

Reply:
👍🏽 I was assuming I'd have to do some amount of rebasing on this patch :)

Reply:
@mfb so what we can do is switch the temp tables to a construct like
```
CRM_Utils_SQL_TempTable::build()->createWithQuery($sql)
```

as preliminary clean up & then we just have one place to intervene

Reply:
(although skip this particular one as we'll get rid of it instead)

Reply:
@mfb ALSO https://github.com/eileenmcnaughton/civicrm-core/blob/6b83d5bdd0f2ca546924feae6aa42aeddb1d40cf/CRM/Utils/SQL/TempTable.php#L97 suggests some possibilities

Reply:
I saw that line but didn't dig in and see what it means. Technically it's not related, I'm just trying to migrate everything that is already utf8 to utf8mb4 :)

Reply:
right - but it's one place that could pick up the change for all temp tables & if it doesn't feel safe we can use a constant like that for grandfathering

Reply:
This line as well as CRM/Contact/Form/Search/Custom/FullText.php is about MEMORY (HEAP) tables (shared between sessions).  Meanwhile it looks like CRM/Utils/SQL/TempTable.php is about no-engine-specified (defaults to InnoDB) TEMPORARY tables, which are different (session-specific).  

I'd agree all the queries re: memory and temporary tables could be cleaned up/refactored, but I'd say it should happen in a focused PR?

Reply:
Hmm - I'm  a bit fan of picking things off in small easy to review PRs so when the PR that actually does the change happens it's not so daunting it gets ignored.

Reply:
Ah I just realized all the MEMORY (HEAP) tables are temporary, I didn't realize that.   So, I guess the main problem is that CRM/Utils/SQL/TempTable.php doesn't support creating MEMORY tables.

Reply:
I'm assuming that could just be added in there - maybe see if @totten has any thoughts first..

Reply:
I created a new PR #13644 for refactoring temporary table stuff, and resolved this discussion.